### DISCUSSION TITLE
Server new messages notifications on Android(K-9 and Fairemail) and Windows (Thunderbird)

### DISCUSSION THREAD

**Discussion by BradNagel** [forwardemail/forwardemail.net#284] [Q&A] [answered]
Wanted to ask about new message notifications because maybe I'm missing something and this is expected behavior. 

While testing applications to figure out what I wanted to use on my desktop and phone I noticed when I sent a email to my mailbox that new notifications were kinda of random or not at all till the periodical 15 min sync happened. What I found is that if any one of the three applications currently running did a sync then very shortly after all of the other email applications would display "New Message" notifications. Its almost like the trigger for a new email was depending on one of the clients to do a sync before it would trigger to the other clients. I can move emails marked as unread from the trash to the inbox on one client and the other clients will be able to see that movement perfectly and notify me of a new email. So what trigger is missing when a email is sent to my mailbox externally where nothing is passed to the clients?

**Comment by graemev** [comment]
You probably need to say something about **how** you are getting messages (pop, imap etc) ...The interaction between clients makes me think of [IMAP Idle](https://en.wikipedia.org/wiki/IMAP_IDLE)  may be involved.

https://support.mozilla.org/en-US/kb/configuration-options-accounts  (search idle)

**Reply by BradNagel** [reply]
Yes, It is IMAP and IMAP IDLE does seem to be enabled and working for most cases unless it's a new external email.

**Reply by graemev** [reply]
Opps, didn't see "reply" link (I think I posted as an answer)

Well you can see how you moving messages from one mailbox to another via IMAP would cause the IMAP IDLE to get woken up  (developers of IMAP interface are hardly likely to miss it :-) )

So this could be the forwardemail implementation of SMTP is not honouring the IDLE.

I had seen a big delay from a (test) mail getting sent and the IMAP IDLE waking up. I'd put it down to "just being slow" but it may well be , as you suggest, not until some other activity.

One thing I had noticed. If I send a test mail (to a test mailbox), wait an hour then IMAP Poll it, it's not there. If I then immediately IMAP poll it again , it appears. Since it's unlikely to, consistently , just arrive in that window, I take it that it takes 2 polls to get a new message. I reported this as a bug many months ago but I've seen nothing about it.

AFYI. They reboot the servers every hour . I see in my logs the client sitting on "idle" gets an error drops the connection and has to make a new one (to the new server)  ...I guess this event might also "wake it up".

**Reply by titanism** [reply, marked as answer]
Our IMAP server does a reload now once every 12 hours instead of 4 to refresh memory.  We're going to test and if all goes well we will turn off reloads completely.  These issues should be resolved now.  If you continue to have errors or slowness let us know of the specific error code you receive (e.g. connection timeout) and as much detail/screenshots you can provide (email works best, send direct to `support@forwardemail.net`).
